src/assets/TokenBalancesController.test.ts

```diff
@@ -182,21 +182,28 @@ describe('TokenBalancesController', () => {
     const assetsContract = new AssetsContractController();
     const network = new NetworkController();
     const preferences = new PreferencesController();
-    const assets = new TokensController({
+    const tokensController = new TokensController({
       onPreferencesStateChange: (listener) => preferences.subscribe(listener),
       onNetworkStateChange: (listener) => network.subscribe(listener),
     });
+
+    const supportsInterfaceStub = stub().returns(Promise.resolve(false));
+
+    stub(tokensController, '_createEthersContract').callsFake(() =>
```

Comment:
This isn't ideal :thinking: Now the tests are tied into controller internals.

Like the APIs, I had hoped we would extract network interactions into separate modules so they could be more easily mocked in tests without needing to understand how the controller worked internally. We don't have any examples of this yet though. I don't know how to handle the provider, since unlike the APIs it is stateful.

Reply:
Yeah I see what you mean. I'm not sure how to handle the provider here either. Maybe we can put our heads together on a call to think this through?

Reply:
Sure! Not a blocker for this PR, but we'll want to figure out a pattern for how to do this eventually.

Reply:
How about I document this in a separate ticket as part of a broader refactor to create a pattern for separating network interactions into modules that can be more cleanly mocked/stubbed?

Reply:
Sure! Or rather, just add it to this list: https://github.com/MetaMask/controllers/issues/430 There is already a ticket for it.

Reply:
Done!